Firestore/core/src/firebase/firestore/model/field_transform.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_FIELD_TRANSFORM_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_FIELD_TRANSFORM_H_
+
+#include <memory>
+#include <utility>
+
+#include "Firestore/core/src/firebase/firestore/model/field_path.h"
+#include "Firestore/core/src/firebase/firestore/model/transform_operations.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+/** A field path and the TransformOperation to perform upon it. */
+class FieldTransform {
+ public:
+  FieldTransform(FieldPath path,
+                 std::unique_ptr<TransformOperation> transformation)
+      : path_(std::move(path)), transformation_(std::move(transformation)) {
+  }
+
+  const FieldPath& path() const {
+    return path_;
+  }
+
+  const TransformOperation* transformation() const {
```

Comment:
I realize that inside `FieldTransform` we need the `TransformOperation` to be stored in a `unique_ptr` so that we can store different subtypes. However, it seems like this implementation detail does not need to leak outside `FieldTransform` and this method could return a `const TransformOperation&` instead of a const pointer.

Reply:
That's true. Fixed.